File: backend/app/models/crop.py
from sqlalchemy import Column,Integer,String,Date
from app.database import Base


class Crop(Base):

    __tablename__="crops"


    id = Column(Integer,primary_key=True)

    farmer_id = Column(Integer)

    crop_name = Column(String)

    quantity = Column(Integer)

    harvest_date = Column(Date)

File: src/lib/server/services/crop.ts
/** Business logic for crop inventory management. */

import type { Crop } from "../models";
import { crops, farmers } from "../repositories";
import { NotFoundError } from "./errors";

export interface CropInput {
  farmer_id: number;
  crop_name: string;
  quantity: number;
  harvest_date: string | null;
}

export function listCrops(options: { skip?: number; limit?: number }): Crop[] {
  return crops.list(options);
}

export function getCrop(cropId: number): Crop | null {
  return crops.get(cropId);
}

export function createCrop(input: CropInput): Crop {
  if (farmers.get(input.farmer_id) === null) {
    throw new NotFoundError(`Farmer ${input.farmer_id} does not exist`);
  }
  return crops.create(input);
}
